### PR TITLE
07762 - Treat All SSLExceptions as SOCKET_EXCEPTIONs

### LINKED ISSUE #7762: Unsafe closing of sockets
The issue seems to be that when we receive a new connection from a peer, we close it while the negotiator is still using it. This needs to be handled more elegantly.

```
2023-07-31 13:19:51.107 131844   ERROR SOCKET_EXCEPTIONS <<sync_server #426>> InboundConnectionManager: 2 got new connection from 1, disconnecting old one
2023-07-31 13:19:52.871 131845   ERROR SOCKET_EXCEPTIONS <<platform-core: SyncProtocolWith0 2 to 0>> NetworkUtils: Connection broken: 2 <- 0 Caused By: {Exception: IOException Message: com.swirlds.common.threading.pool.ParallelExecutionException: Time thrown: 2023-07-31T13:19:52.871700100Z Caused By: {Exception: ParallelExecutionException Message: Time thrown: 2023-07-31T13:19:52.871700100Z Caused By: {Exception: EOFException Message: null }}}
2023-07-31 13:19:52.872 131846   ERROR EXCEPTION        <<platform-core: SyncProtocolWith1 2 to 1>> NetworkUtils: Connection broken: 2 <- 1
javax.net.ssl.SSLException: Cannot read field "processed" because "this.engine" is null
	at sun.security.ssl.Alert.createSSLException(Alert.java:133) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:371) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:314) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:309) ~[?:?]
	at sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1702) ~[?:?]
	at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1075) ~[?:?]
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:244) ~[?:?]
	at java.io.BufferedInputStream.read(BufferedInputStream.java:263) ~[?:?]
	at com.swirlds.common.io.extendable.extensions.AbstractStreamExtension.read(AbstractStreamExtension.java:75) ~[swirlds-common-0.40.2.jar:?]
	at com.swirlds.common.io.extendable.ExtendableInputStream.read(ExtendableInputStream.java:70) ~[swirlds-common-0.40.2.jar:?]
	at com.swirlds.common.io.extendable.extensions.AbstractStreamExtension.read(AbstractStreamExtension.java:75) ~[swirlds-common-0.40.2.jar:?]
	at com.swirlds.common.io.extendable.ExtendableInputStream.read(ExtendableInputStream.java:70) ~[swirlds-common-0.40.2.jar:?]
	at java.io.FilterInputStream.read(FilterInputStream.java:82) ~[?:?]
	at com.swirlds.common.io.streams.AugmentedDataInputStream.read(AugmentedDataInputStream.java:71) ~[swirlds-common-0.40.2.jar:?]
	at com.swirlds.platform.network.communication.states.SentKeepalive.transition(SentKeepalive.java:59) ~[swirlds-platform-core-0.40.2.jar:?]
	at com.swirlds.platform.network.communication.Negotiator.execute(Negotiator.java:94) ~[swirlds-platform-core-0.40.2.jar:?]
	at com.swirlds.platform.network.communication.NegotiatorThread.run(NegotiatorThread.java:74) ~[swirlds-platform-core-0.40.2.jar:?]
	at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.doWork(StoppableThreadImpl.java:622) ~[swirlds-common-0.40.2.jar:?]
	at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.run(StoppableThreadImpl.java:217) ~[swirlds-common-0.40.2.jar:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot read field "processed" because "this.engine" is null
	at com.sun.crypto.provider.GaloisCounterMode.checkDataLength(GaloisCounterMode.java:578) ~[?:?]
	at com.sun.crypto.provider.GaloisCounterMode$GCMDecrypt.doFinal(GaloisCounterMode.java:1433) ~[?:?]
	at com.sun.crypto.provider.GaloisCounterMode.engineDoFinal(GaloisCounterMode.java:447) ~[?:?]
	at javax.crypto.Cipher.doFinal(Cipher.java:2500) ~[?:?]
	at sun.security.ssl.SSLCipher$T12GcmReadCipherGenerator$GcmReadCipher.decrypt(SSLCipher.java:1659) ~[?:?]
	at sun.security.ssl.SSLSocketInputRecord.decodeInputRecord(SSLSocketInputRecord.java:260) ~[?:?]
	at sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:181) ~[?:?]
	at sun.security.ssl.SSLTransport.decode(SSLTransport.java:111) ~[?:?]
	at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1503) ~[?:?]
	at sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1474) ~[?:?]
	at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1059) ~[?:?]
	... 14 more
```

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:
Adds a condition to the NetworkUtils to classify any SSLException as a SOCKET_EXCEPTION even if the SSLException has a nested cause.  

**Related issue(s)**:
Fixes #7762

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Utilities.java [hunk 1]
```diff
@@ -330,4 +330,25 @@ public static boolean isRootCauseSuppliedType(final Throwable t, final Class<? e
...
+    /**
+     * Checks all nesting of causes for any instance of the supplied type.
+     *
+     * @param t
+     * 		the throwable to unwrap
+     * @param type
+     * 		the type to check against
+     * @return true if any of the causes matches the supplied type, false otherwise.
+     */
+    public static boolean hasAnyCauseSuppliedType(final Throwable t, final Class<? extends Throwable> type) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Utilities.java:343]: missing nullity annotations

**Review by alittley — APPROVED** [review]

**Review by cody-littley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Utilities.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Utilities.java]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java [hunk 2]
```diff
@@ -89,7 +90,9 @@ public static void handleNetworkException(final Exception e, final Connection co
      * @return the marker to use for logging
      */
     public static Marker determineExceptionMarker(final Exception e) {
-        return Utilities.isCausedByIOException(e) || Utilities.isRootCauseSuppliedType(e, SyncTimeoutException.class)
+        return Utilities.isCausedByIOException(e)
+                        || Utilities.isRootCauseSuppliedType(e, SyncTimeoutException.class)
+                        || Utilities.hasAnyCauseSuppliedType(e, SSLException.class)
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java:95]: add a comment why we added this, with a link to the ticket

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java:97]: added

**Pull request merged**